runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/EngineManager.java

```diff
@@ -421,31 +421,33 @@ private void unregister(
     private final class NameResolver
     {
         private final int namespaceId;
-        private final Matcher matchName;
+        private final ThreadLocal<Matcher> matchName;
 
         private NameResolver(
             int namespaceId)
         {
             this.namespaceId = namespaceId;
-            this.matchName = NamespaceAdapter.PATTERN_NAME.matcher("");
+            this.matchName = new ThreadLocal<>();
         }
 
         private long resolve(
             String name)
         {
             long id = 0L;
-
-            if (name != null && matchName.reset(name).matches())
+            if (name != null)
```

Comment:
```suggestion
            Matcher matchName = this.matchName.get();
            if (name != null && matchName.reset(name).matches())
```
...plus revert other changes to this method.